### PR TITLE
TextTrackCueList

### PR DESCRIPTION
**Pull request by rachelandrew**
Adding TextTrackCueList.

reviewer: @jpmedley 

The other WebVTT interfaces are all fairly sparsely documented, mostly just the interface pages and no examples.

### CODE REVIEW

File: files/en-us/web/api/texttrackcuelist/getcuebyid/index.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  <dd>A {{domxref("DOMString")}} which is an identifier for the cue.</dd>
+</dl>
+
+<h3 id="Returns">Return value</h3>
+
+<p>A {{domxref("VTTCue")}} object.</p>
+
+<h2 id="Examples">Examples</h2>
+
+<p>The {{domxref("TextTrack.cues")}} read-only property then returns a {{domxref("TextTrackCueList")}} containing the current cues for that particular track. Calling <code>cues.getCueById("second")</code> returns the {{domxref("VTTCue")}} with an ID of "second".</p>
```
Comment (jpmedley) [inline files/en-us/web/api/texttrackcuelist/getcuebyid/index.html:34]:
```suggestion
<p>The {{domxref("TextTrack.cues")}} property returns a {{domxref("TextTrackCueList")}} containing the current cues for that particular track. Calling <code>cues.getCueById("second")</code> returns the {{domxref("VTTCue")}} with an ID of "second".</p>
```

Comment (jpmedley) [reply files/en-us/web/api/texttrackcuelist/getcuebyid/index.html:34]: Is there an argument for repeating 'read-only'?

File: files/en-us/web/api/texttrackcuelist/index.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  - Reference
+  - TextTrackCueList
+  - WebVTT
+  - Media
+---
+<div>{{APIRef("WebVTT")}}</div>
+
+<p class="summary">The <strong><code>TextTrackCueList</code></strong> interface represents a dynamically updating list of {{domxref("TextTrackCue")}} objects.</p>
+
+<p>This interface has no constructor.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/texttrackcuelist/index.html:16]: How do I get an instance of it?

File: files/en-us/web/api/texttrackcuelist/index.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+<h2 id="Methods">Methods</h2>
+
+<dl>
+	<dt>{{domxref('TextTrackCueList.getCueById()')}}</dt>
+	<dd>Returns the first {{domxref('TextTrackCue')}} object with the identifier passed to it.</dd>
+</dl>
+
+<h2 id="Examples">Examples</h2>
+
+<p>The {{domxref("HTMLMediaElement.textTracks")}} read-only property returns a {{domxref("TextTrackList")}} object listing all of the {{domxref("TextTrack")}} objects, one for each text track linked to the media. The {{domxref("TextTrack.cues")}} read-only property then returns a <code>TextTrackCueList</code> containing the cues for that particular track.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/texttrackcuelist/index.html:34]:
```suggestion
<p>The {{domxref("HTMLMediaElement.textTracks")}} property returns a {{domxref("TextTrackList")}} object listing all of the {{domxref("TextTrack")}} objects, one for each text track linked to the media. The {{domxref("TextTrack.cues")}} property then returns a <code>TextTrackCueList</code> containing the cues for that particular track.</p>
```

File: files/en-us/web/api/texttrackcuelist/length/index.html [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+<h2 id="Syntax">Syntax</h2>
+
+<pre class="syntaxbox notranslate">var <var>length</var> = <var>TextTrackCueList</var>.length;</pre>
+
+<h3>Value</h3>
+<p>An <code>unsigned long</code> which is the number of cues in the list.</p>
+
+<h2 id="Examples">Examples</h2>
+
+<p>The {{domxref("TextTrack.cues")}} read-only property returns a {{domxref("TextTrackCueList")}} containing the current cues for that particular track. Calling <code>cues.length</code> returns the number of cues in the list. Using the WebVTT track below, the value of <code>length</code> is 5.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/texttrackcuelist/length/index.html:26]:
```suggestion
<p>The {{domxref("TextTrack.cues")}} property returns a {{domxref("TextTrackCueList")}} containing the current cues for that particular track. Calling <code>cues.length</code> returns the number of cues in the list. Using the WebVTT track below, the value of <code>length</code> is 5.</p>
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/texttrackcuelist/getcuebyid/index.html (hunk 1 above)
Comment (rachelandrew) [reply files/en-us/web/api/texttrackcuelist/getcuebyid/index.html:34]: probably not :)

File: files/en-us/web/api/texttrackcuelist/index.html (hunk 2 above)
Comment (rachelandrew) [reply files/en-us/web/api/texttrackcuelist/index.html:16]: Only via other methods that return it - which I've tried to show in the examples, should I have added something to the prose to indicate that? I looked at a couple of other list interfaces and didn't see any common pattern.

Comment (Ryuno-Ki) [reply files/en-us/web/api/texttrackcuelist/index.html:16]: Ah, so a factory pattern?

Comment (jpmedley) [reply files/en-us/web/api/texttrackcuelist/index.html:16]: Please add some prose. The only place I can find it is from `TextTrack.cues`. Please link to that.

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/texttrackcuelist/index.html (hunk 2 above)
Comment (rachelandrew) [reply files/en-us/web/api/texttrackcuelist/index.html]: Have done, this is also what the example shows.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**